src/test/surefire.go

```diff
@@ -9,10 +9,10 @@ import (
 )
 
 // CopySurefireXmlFilesToDir copies all the XML test results files into the given directory.
-func CopySurefireXmlFilesToDir(graph *core.BuildGraph, surefireDir string) {
+func CopySurefireXmlFilesToDir(state *core.BuildState, surefireDir string) {
 	outputDirs := make(map[string]struct{})
-	for _, target := range graph.AllTargets() {
-		if target.IsTest && !target.NoTestOutput {
+	for _, target := range state.Graph.AllTargets() {
```

Comment:
possibly this should be something like `state.ExpandOriginalTargets()` rather than iterating the whole graph, which might include targets that you didn't ask it to test?